lambdas/services/process_mns_message_service.py

```diff
@@ -0,0 +1,150 @@
+from datetime import datetime
+
+from botocore.exceptions import ClientError
+from enums.death_notification_status import DeathNotificationStatus
+from enums.metadata_field_names import DocumentReferenceMetadataFields
+from enums.mns_notification_types import MNSNotificationTypes
+from enums.patient_ods_inactive_status import PatientOdsInactiveStatus
+from models.mns_sqs_message import MNSSQSMessage
+from services.base.dynamo_service import DynamoDBService
+from services.base.sqs_service import SQSService
+from utils.audit_logging_setup import LoggingService
+from utils.exceptions import PdsErrorException
+from utils.utilities import get_pds_service
+
+logger = LoggingService(__name__)
+
+
+class MNSNotificationService:
+    def __init__(self):
+        self.dynamo_service = DynamoDBService()
+        self.table = os.getenv("LLOYD_GEORGE_DYNAMODB_NAME")
+        self.pds_service = get_pds_service()
+        self.sqs_service = SQSService()
+        self.queue = os.getenv("MNS_NOTIFICATION_QUEUE_URL")
+
+    def handle_mns_notification(self, message: MNSSQSMessage):
+        try:
+            if message.type == MNSNotificationTypes.CHANGE_OF_GP:
+                logger.info("Handling GP change notification.")
+                self.handle_gp_change_notification(message)
+
+            elif message.type == MNSNotificationTypes.DEATH_NOTIFICATION:
+                logger.info("Handling death status notification.")
+                self.handle_death_notification(message)
+
+        except PdsErrorException:
+            logger.info("An error occurred when calling PDS")
+            self.send_message_back_to_queue(message)
+
+        except ClientError as e:
+            logger.info(
+                f"Unable to process message: {message.id}, of type: {message.type}"
+            )
+            logger.info(f"{e}")
+
+    def handle_gp_change_notification(self, message: MNSSQSMessage):
+        patient_document_references = self.get_patient_documents(
+            message.subject.nhs_number
+        )
+
+        if not self.patient_is_present_in_ndr(patient_document_references):
+            return
+
+        updated_ods_code = self.get_updated_gp_ods(message.subject.nhs_number)
+
+        for reference in patient_document_references:
+            if reference["CurrentGpOds"] is not updated_ods_code:
+                self.dynamo_service.update_item(
+                    table_name=self.table,
+                    key=reference["ID"],
+                    updated_fields={
+                        DocumentReferenceMetadataFields.CURRENT_GP_ODS.value: updated_ods_code,
+                        DocumentReferenceMetadataFields.LAST_UPDATED.value: int(
+                            datetime.now().timestamp()
+                        ),
+                    },
+                )
+
+        logger.info("Update complete for change of GP")
+
+    def handle_death_notification(self, message: MNSSQSMessage):
+        if self.is_informal_death_notification(message):
```

Comment:
Similar to the above feedback, I'd suggest getting rid of the `is_informal_death_notification` and `is_formal_death_notification` style functions and just changing this to a `match/case` statement on `message.data["deathNotificationStatus"]` for conciseness. We've got a bunch of statements in the style of
```
if X function is true -> return
if Y function is true -> do this
if Z function is true -> do this
```

And I think we could simplify this to just be (very pseudo-code-y sorry!)
```
Match message.data["deathNotificationStatus"]
    case  DeathNotificationStatus.INFORMAL -> do this
    case  DeathNotificationStatus.FORMAL -> do this
    case DeathNotificationStatus.REMOVED -> do this
``` 

This, with the above feedback means that we could massively simplify the `handle_death_notification` function. Please drop me a message if you'd like any clarification.